framework/src/st_things/things_stack/utils/things_ping.c

```diff
@@ -628,7 +628,7 @@ static int is_ip_key_equal(things_ping_s *ping, const char *targetIP)
 		return res;
 	}
 
-	if (strncmp(ping->addr, targetIP, strlen(targetIP)) == 0) {
+	if (strncmp(ping->addr, targetIP, sizeof(targetIP)) == 0) {
 		THINGS_LOG_D(TAG, "Found ping-struct(%s) for targetIP(%s)", ping->addr, targetIP);
```

Comment:
sizeof(targetIP) is always 4. because it's pointer.
revert to strlen.

Reply:
i fixed